docs/en/integrations/dbt/dbt-setup.md

```diff
@@ -0,0 +1,165 @@
+---
+sidebar_label: Setup 
+sidebar_position: 2
+description: Setup of dbt and the ClickHouse plugin
+---
+
+# Installation
+
+## dbt
+
+We assume the use of the dbt CLI for the following examples. Users may also wish to consider[ dbt Cloud](https://docs.getdbt.com/docs/dbt-cloud/cloud-overview), which offers a web-based Integrated Development Environment (IDE) allowing users to edit and run projects.
+
+dbt offers a number of options for CLI installation. Follow the instructions described[ here](https://docs.getdbt.com/dbt-cli/install/overview). At this stage install dbt-core only. We recommend the use of `pip`.
+
+```bash
+python install dbt-core
+```
+
+**Important: The following is tested under python 3.9.**
+
+## ClickHouse plugin
+
+Install the dbt ClickHouse plugin:
+
+```bash
+pip install dbt-clickhouse
+```
+
+# Prepare ClickHouse
+
+dbt excels when modeling highly relational data. For the purposes of example, we provide a small IMDB dataset with the following relational schema. This dataset originates from the[ relational dataset repository](https://relational.fit.cvut.cz/dataset/IMDb). This is trivial relative to common schemas used with dbt but represents a manageable sample:
+
+
+<img src={require('./images/dbt_01.png').default} class="image" alt="IMDB table schema" style={{width: '100%'}}/>
+
+We use a subset of these tables as shown. 
+
+Create the following table engines:
+
+
+```sql
+CREATE DATABASE imdb ENGINE=Atomic;
+
+CREATE TABLE imdb.actors
+(
+    id         UInt32,
+    first_name String,
+    last_name  String,
+    gender     FixedString(1)
+) ENGINE = MergeTree ORDER BY (id, first_name, last_name, gender);
+
+CREATE TABLE imdb.directors
+(
+    id         UInt32,
+    first_name String,
+    last_name  String
+) ENGINE = MergeTree ORDER BY (id, first_name, last_name);
+
+CREATE TABLE imdb.genres
+(
+    movie_id UInt32,
+    genre    String
+) ENGINE = MergeTree ORDER BY (movie_id, genre);
+
+CREATE TABLE imdb.movie_directors
+(
+    director_id UInt32,
+    movie_id    UInt64
+) ENGINE = MergeTree ORDER BY (director_id, movie_id);
+
+CREATE TABLE imdb.movies
+(
+    id   UInt32,
+    name String,
+    year UInt32,
+    rank Float32 DEFAULT 0
+) ENGINE = MergeTree ORDER BY (id, name, year);
+
+CREATE TABLE imdb.roles
+(
+    created_at DateTime DEFAULT now(),
+    actor_id   UInt32,
+    movie_id   UInt32,
+    role       String
+) ENGINE = MergeTree ORDER BY (actor_id, movie_id);
+```
+
+Note the column created_at for the table roles, which defaults to a value of now(). We use this later to identify incremental updates to our models - see [Incremental Models](./dbt-incremental-model).
+
+We use the s3 function to read the source data from public endpoints to insert data into these. Use the following to populate the tables:
+
+```sql
+INSERT INTO imdb.actors
```

Comment:
The S3 repositories are not publicly available
```
Received exception from server (version 22.4.1):
Code: 499. DB::Exception: Received from localhost:9000. DB::Exception: Access Denied: While executing TabSeparatedRowInputFormat: While executing S3. (S3_ERROR)
```

Reply:
resolved